src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -791,7 +791,20 @@ private TypeSymbol BindVariableType(CSharpSyntaxNode declarationNode, Diagnostic
             associatedSyntaxNode = associatedSyntaxNode ?? declarator;
 
             // Check for variable declaration errors.
-            bool hasErrors = this.ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
+            Binder nameConflictChecker = this;
+
+            // Step out of the PatternVariableBinder for locals declared in variable declaration statement
+            if (this is PatternVariableBinder)
+            {
+                CSharpSyntaxNode parent;
+                if ((parent = declarator.Parent)?.Kind() == SyntaxKind.VariableDeclaration &&
```

Comment:
@gafter This function is called for declaration in many statements: using, for, catch, local declaration, ... My intent is to special case only local declarations. In addition to local declarations , only 'using' statement is tested. However, I believe other statements will share scope between their locals and pattern locals. Local declaration statement is special because regular locals, declared in it, belong to a different scope, by comparison to pattern locals, declared in it. Does this address your concern?